Microsoft.Toolkit.Uwp.UI.Controls.DataGrid/DataGrid/DataGridComboBoxColumn.cs

```diff
@@ -0,0 +1,547 @@
+            }
+
+            if (_fontSize.HasValue)
+            {
+                comboBox.FontSize = _fontSize.Value;
+            }
+
+            if (_fontStyle.HasValue)
+            {
+                comboBox.FontStyle = _fontStyle.Value;
+            }
+
+            if (_fontWeight.HasValue)
+            {
+                comboBox.FontWeight = _fontWeight.Value;
+            }
+
+            RefreshForeground(comboBox, (cell != null & cell.OwningRow != null) ? cell.OwningRow.ComputedForeground : null);
+
+            comboBox.SelectionChanged += (sender, args) =>
+            {
+                var item = args.AddedItems.FirstOrDefault();
+                if (item != null)
+                {
+                    var newValue = DisplayMemberPath != null
+                        ? item.GetType().GetProperty(Binding.Path.Path).GetValue(item)
+                        : item;
+
+                    dataItem.GetType().GetProperty(Binding.Path.Path).SetValue(dataItem, newValue);
+                }
+            };
+
+            return comboBox;
+        }
+
+        /// <summary>
+        /// Gets a read-only <see cref="T:Windows.UI.Xaml.Controls.TextBlock"/> element that is bound to the column's <see cref="P:Microsoft.Toolkit.Uwp.UI.Controls.DataGridBoundColumn.Binding"/> property value.
+        /// </summary>
+        /// <param name="cell">The cell that will contain the generated element.</param>
+        /// <param name="dataItem">The data item represented by the row that contains the intended cell.</param>
+        /// <returns>A new, read-only <see cref="T:Windows.UI.Xaml.Controls.TextBlock"/> element that is bound to the column's <see cref="P:Microsoft.Toolkit.Uwp.UI.Controls.DataGridBoundColumn.Binding"/> property value.</returns>
+        protected override FrameworkElement GenerateElement(DataGridCell cell, object dataItem)
+        {
+            var textBlockElement = new TextBlock
+            {
+                Margin = new Thickness(DATAGRIDCOMBOBOXCOLUMN_leftMargin, 0.0, DATAGRIDCOMBOBOXCOLUMN_rightMargin, 0.0),
+                VerticalAlignment = VerticalAlignment.Center
+            };
+
+            if (DependencyProperty.UnsetValue != ReadLocalValue(DataGridComboBoxColumn.FontFamilyProperty))
+            {
+                textBlockElement.FontFamily = this.FontFamily;
+            }
+
+            if (_fontSize.HasValue)
+            {
+                textBlockElement.FontSize = _fontSize.Value;
+            }
+
+            if (_fontStyle.HasValue)
+            {
+                textBlockElement.FontStyle = _fontStyle.Value;
+            }
+
+            if (_fontWeight.HasValue)
+            {
+                textBlockElement.FontWeight = _fontWeight.Value;
+            }
+
+            RefreshForeground(textBlockElement, (cell != null & cell.OwningRow != null) ? cell.OwningRow.ComputedForeground : null);
+
+            bool isDesignMode = Windows.ApplicationModel.DesignMode.DesignModeEnabled;
```

Comment:
While doing a bit of testing I realized there is a bug in the textbox column type. Sorry about that. Maybe the Silverlight framework allowed FrameworkElement.SetBinding(<prop>, null) but the UWP XAML framework does not. So when this.Binding is null, the app crashes.
It seems the condition here should be:
if (this.Binding != null && this.DisplayMemberPath == null && EnsureOwningGrid()) 
{
    ...
}